### PR TITLE
Replace packr and packd in gobuffalo cli with embed.FS

### PR DESCRIPTION
**Pull request by fasmat**
I started migrating a bunch of files away from `packr` to `embed` already, but some are still missing. This PR is just here to keep track of the changes and allow for early feedback.

I needed to upgrade `gobuffalo/genny` to be compatible with `io/fs` and other golang 1.16 features: https://github.com/gobuffalo/genny/pull/42

Next steps:
- the `gobuffalo/buffalo` codebase also needs to be migrated from `packr` to `embed.FS`
- update cli to new gobuffalo codebase:
  - update templates to fit new codebase
  - update internal coke project (used in tests) to newly generated templates
  - add fix routines to aid in migration

### CODE REVIEW

**Comment by fasmat** [comment]
The build currently fails because in 1.16 files and directories starting with a dash (e.g. `-dot-env.tmpl`) cannot be embedded. This seems to have been fixed in 1.17 (although files starting with `_` or `.` still can only be embedded when listed explicitly in the glob for the //go:embed directive).

I don't know how to get around this issue without making some bigger changes to genny, but I also don't want to wait until Feb 2022 until gobuffalo drops support for 1.16 to finally finish the migration from packr to embed.FS. What is your opinion @paganotoni?

The next steps:

- [ ] Update coke project to most recent code base
- [ ] Migrate gobuffalo/buffalo from packr to embed.FS
- [ ] Add fix routines to help current users of gobuffalo with the migration
- [ ] Add a few missing tests (see comments I added)
- [ ] Code Review
- [ ] Decide on how to deal with the 1.16 limitation of embed.FS

File: internal/genny/build/_fixtures/coke/go.mod [hunk 1]
```diff
@@ -0,0 +1,100 @@
+module coke
+
+go 1.17
```
Comment (paganotoni) [inline internal/genny/build/_fixtures/coke/go.mod:3]: Would this work when building against Go 1.16 (We still run tests there).

File: internal/genny/docker/docker.go [hunk 2]
```diff
@@ -19,19 +24,51 @@ func New(opts *Options) (*genny.Generator, error) {
...
+	}
+
+	if err := g.FS(common); err != nil {
+		return g, err
+	}
 
 	switch opts.Style {
 	case "multi":
-		g.Box(packr.New("github.com/gobuffalo/buffalo/multi", "../docker/templates/multi"))
+		multi, err := fs.Sub(templates, "templates/multi")
```
Comment (paganotoni) [inline internal/genny/docker/docker.go:38]: There is a follow up action from this one: moving to only have multi-stage docker file. That would simplify this generator.

**Review by paganotoni — APPROVED** [review]
LGTM @fasmat, thanks for putting this together. Left a couple of comments on things that would be nice to have and future tasks. It was a nice excercise to review this PR.

File: internal/genny/build/_fixtures/coke/go.mod (hunk 1 above)
Comment (fasmat) [reply internal/genny/build/_fixtures/coke/go.mod:3]: Yes, this is just a newly generated project with cli and it defaults to the installed go version (1.17 in my setup).

I will update that again when gobuffalo/buffalo is completely free of packed and then tackle the Todo that mentioned generating and building a new buffalo project.

File: internal/genny/docker/docker.go (hunk 2 above)
Comment (fasmat) [reply internal/genny/docker/docker.go:38]: Yes I've seen https://github.com/gobuffalo/cli/issues/42 but we should resolve that task in a dedicated PR. I didn't want to do too much at once.

Comment (paganotoni) [reply internal/genny/docker/docker.go:38]: Yes, I'm on that one.

**Pull request merged**